test/unit/org/apache/cassandra/schema/createlike/CreateLikeTest.java

```diff
@@ -0,0 +1,545 @@
+        assertRows(execute("SELECT * FROM " + targetKs + "." + targetTb),
+                   row(2, d2, (byte)5, f2, list("c", "d"), map("nk", 2), duration2));
+
+        sourceTb = createTable(sourceKs, "CREATE TABLE %s (a int , " +
+                                                                "b text, " +
+                                                                "c bigint, " +
+                                                                "d decimal, " +
+                                                                "e set<text>, " +
+                                                                "f uuid, " +
+                                                                "g vector<int, 2>, " +
+                                                                "h list<float>, " +
+                                                                "i timeuuid, " +
+                                                                "j map<text, frozen<set<int>>>, " +
+                                                                "PRIMARY KEY((a, b), c, d)) " +
+                                                                "WITH CLUSTERING ORDER BY (c DESC, d ASC);");
+        targetTb = createTableLike("CREATE TABLE %s LIKE %s", sourceTb, sourceKs, targetKs);
+        assertTableMetaEquals(sourceKs, targetKs, sourceTb, targetTb);
+        execute("INSERT INTO " + sourceKs + "." + sourceTb + " (a, b, c, d, e, f, g, h, i, j)  VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                1, "b", 100L, decimal1, set("1", "2"), uuid1, vector1, list(1.1F, 2.2F), timeUuid1, map("k", set(1, 2)));
+        execute("INSERT INTO " + targetKs + "." + targetTb + " (a, b, c, d, e, f, g, h, i, j) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                2, "nb", 200L, decimal2, set("3", "4"), uuid2, vector2, list(3.3F, 4.4F), timeUuid2, map("nk", set(3, 4)));
+        assertRows(execute("SELECT * FROM " + sourceKs + "." + sourceTb),
+                   row(1, "b", 100L, decimal1, set("1", "2"), uuid1, vector1, list(1.1F, 2.2F), timeUuid1, map("k", set(1, 2))));
+        assertRows(execute("SELECT * FROM " + targetKs + "." + targetTb),
+                   row(2, "nb", 200L, decimal2, set("3", "4"), uuid2, vector2, list(3.3F, 4.4F), timeUuid2, map("nk", set(3, 4))));
+    }
+
+    @Test
+    public void testIfNotExists() throws Throwable
+    {
+        String sourceTb = createTable(sourceKs, "CREATE TABLE %s (a int, b text, c duration, d float, PRIMARY KEY(a, b));");
+        String targetTb = createTableLike("CREATE TABLE IF NOT EXISTS %s LIKE %s", sourceTb, sourceKs, targetKs);
+        assertTableMetaEquals(sourceKs, targetKs, sourceTb, targetTb);
+
+        createTableLike("CREATE TABLE IF NOT EXISTS %s LIKE %s", sourceTb, sourceKs, targetTb, targetKs);
+        assertInvalidThrowMessage("Cannot add already existing table \"" + targetTb + "\" to keyspace \"" + targetKs + "\"", AlreadyExistsException.class,
```

Comment:
@Maxwell-Guo it should not behave like this. "CREATE TABLE IF NOT EXISTS" is not supposed to throw exception in case a table exists already. You can try it on normal "CREATE TABLE", not dealing with "LIKE".

When the table is created already, then nothing happens.

Reply:
@Maxwell-Guo 

````
CREATE TABLE ks.tb ...
// this will not fail
CREATE TABLE ks.tb_copy FROM ks.tb;
// this will fail because we just created ks_tb_copy on the line above
CREATE TABLE ks.tb_copy FROM ks.tb;
// this will not fail, because ks.tb_copy already exists, so "IF NOT EXISTS" mean that we are not going to create it
CREATE TABLE IF NOT EXISTS ks.tb_copy FROM ks.tb;
````

Reply:
I tested it with your branch

````
$ ./bin/cqlsh
Connected to Test Cluster at 127.0.0.1:9042
[cqlsh 6.3.0 | Cassandra 5.1-SNAPSHOT | CQL spec 3.4.8 | Native protocol v5]
Use HELP for help.
cassandra@cqlsh> CREATE KEYSPACE ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
cassandra@cqlsh> CREATE TABLE ks.tb ( id int PRIMARY KEY );
cassandra@cqlsh> CREATE TABLE ks.tb_copy LIKE ks.tb;
cassandra@cqlsh> CREATE TABLE ks.tb_copy LIKE ks.tb;
AlreadyExists: Table 'ks.tb_copy' already exists
cassandra@cqlsh> CREATE TABLE IF NOT EXISTS ks.tb_copy LIKE ks.tb;
cassandra@cqlsh> 
````

So it all works as expected. Then I do not understand what this test is doing. Maybe I have interpreted it wrong?

Reply:
@Maxwell-Guo could you please add a test like this?

````
CREATE TABLE ks.tb_copy LIKE ks.tb;
CREATE TABLE ks.tb_copy_2 LIKE ks.tb_copy;
````

Basically, test that can create a copy of a copy. 

Please forget this when this is already tested.

Reply:
@Maxwell-Guo ahaaaa! :D I understand now. I have not noticed that there is also this

````
 "CREATE TABLE " + targetKs + "." + targetTb + " LIKE " + sourceKs + "." + sourceTb);
````

so it indeed makes sense. Sorry for the confusion.